### ISSUE TITLE
protocols.tls pydoctor errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5243]
|<img alt="magmatt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| magmatt reported|
|-|-|
|Trac ID|trac#5243|
|Type|defect|
|Created|2011-08-30 14:48:46Z|
|Branch|https://github.com/twisted/twisted/tree/tls-api-links-5243|

pydoctor output:

```
twisted.protocols.tls:0 invalid ref to IProtocol
twisted.protocols.tls:0 invalid ref to IProtocol
twisted.protocols.tls:0 invalid ref to IProtocol
twisted.protocols.tls:0 invalid ref to ITransport
twisted.protocols.tls:0 invalid ref to IReactorSSL
twisted.protocols.tls.TLSMemoryBIOProtocol:207 invalid ref to OpenSSL.SSL.Connection
twisted.protocols.tls.TLSMemoryBIOProtocol:207 invalid ref to ITLSTransport.startTLS
twisted.protocols.tls.TLSMemoryBIOProtocol:207 invalid ref to OpenSSL.SSL.Error
twisted.protocols.tls.TLSMemoryBIOProtocol:207 invalid ref to OpenSSL.SSL.Error
twisted.protocols.tls.TLSMemoryBIOProtocol:207 invalid ref to OpenSSL.SSL.Error
twisted.protocols.tls.TLSMemoryBIOProtocol.getHandle:274 invalid ref to OpenSSL.SSL.Connection
twisted.protocols.tls.TLSMemoryBIOProtocol.getHandle:274 invalid ref to OpenSSL.SSL.Connection
twisted.protocols.tls.TLSMemoryBIOProtocol.getHandle:274 invalid ref to twisted.internet.ssl.Certificate
twisted.protocols.tls.TLSMemoryBIOProtocol.makeConnection:287 invalid ref to OpenSSL.SSL.Connection
```

Attachments:

* [twisted.protocols.tls.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b3d/b3d741708f6cf661e71ab396e47d646952c1a203/731735a5892392f5a89dc39cf50247e5b5a67f82.diff) (4660 bytes) - added by magmatt on 2011-08-30 14:49:27Z - Patch

<details><summary>Searchable metadata</summary>

```
trac-id__5243 5243
type__defect defect
reporter__magmatt magmatt
priority__normal normal
milestone__ 
branch__branches_tls_api_links_5243 branches/tls-api-links-5243
branch_author__magmatt magmatt
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1314715726000000 1314715726000000
changetime__1315833487000000 1315833487000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

Patches need the review keyword so they can be looked at (see ReviewProcess).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [32499]) Branching to 'tls-api-links-5243'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [32500]) Apply twisted.protocols.tls.diff 

refs [#5243](https://github.com/twisted/twisted/issues/5243)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to magmatt<br>@exarkun set status to `new`|
|-|-|

Thanks.  I think two things should change:

  1. It seems like references inside Twisted should either be unqualified (like `IProtocol`) as long as that's valid, otherwise fully qualified (like `twisted.internet.interfaces.IProtocol`), not something in between (like `internet.interfaces.IProtocol`) which works for reasons I don't entirely understand...
  1. I guess `L{OpenSSL.SSL.Connection}` is better than `C{OpenSSL.SSL.Connection}`, if only there were OpenSSL API docs to link to.  Perhaps there `could` be, somehow.  Maybe we should ignore the problem of links to external projects for now, and hope that we can get them to resolve somehow, rather than unlinkifying them all.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

(In [32513]) address review comments, refs [#5243](https://github.com/twisted/twisted/issues/5243)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra removed owner|
|-|-|

Addressed review comments, back up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Thanks.  Looks good to me and to buildbot.  Please add a (misc) news fragment and merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

(In [32528]) news file, refs [#5243](https://github.com/twisted/twisted/issues/5243)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [32529]) Merge tls-api-links-5243: Fixed incorrect epytext markup in docstrings.

Author: magmatt, thijs
Reviewer: exarkun
Fixes: [#5243](https://github.com/twisted/twisted/issues/5243)
